SpringProject/src/main/java/com/gradproject2019/conferences/service/ConferenceServiceImpl.java

```diff
@@ -18,4 +21,12 @@ public ConferenceServiceImpl(ConferenceRepository conferenceRepository) {
         return conferenceRepository.findAll();
     }
 
+    @Override
+    public Conference findConferenceById(Long conferenceId) {
+        Optional<Conference> conference = conferenceRepository.findById(conferenceId);
+        if(conference.isPresent())
```

Comment:
Never use isPresent and get method from Optional. It's a smell and it means that smth is wrong in your method.

Reply:
I disagree. In this case I think it's perfectly fine to use an Optional to handle possible null values from the repository.

Reply:
Repository layer should return Optional as well. As I already said returning null values is a bad practice. That's why in Java 8 Optionals were introduced.

Reply:
We have been trying to research this and haven't found an alternative to the isPresent method, please see linked Oracle page:https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html ,  do you have a suggestion of another method? Thank you